clash-ghc/src-bin-841/Clash/GHCi/UI.hs

```diff
@@ -1998,6 +1998,8 @@ makeHDL backend optsRef srcs = do
                 let dbs = reverse [p | PackageDB (PkgConfFile p) <- packageDBFlags dflags]
                 (bindingsMap,tcm,tupTcm,topEntities,primMap,reprs) <-
                   generateBindings color primDirs idirs dbs hdl src (Just dflags)
+                mainTopEntity <-
+                  sequence (getMainTopEntity src topEntities <$> GHC.mainFunIs dflags)
```

Comment:
```suggestion
                  traverse (getMainTopEntity src topEntities) (GHC.mainFunIs dflags)
```

Reply:
`traverse` because it's more general, or is there something else going on?

Reply:
Because that's what traverse is for, to map an effectful computation over a Functor-like thing.